src/entities/CalcFactor.jl

```diff
@@ -35,15 +35,16 @@ struct CalcFactorNormSq{
   X, 
   C, 
   VT <: Tuple, 
-  M <: AbstractManifold
+  M <: AbstractManifold,
+  MEAS
 } <: CalcFactor{FT}
   """ the interface compliant user object functor containing the data and logic """
   factor::FT
   """ what is the sample (particle) id for which the residual is being calculated """
   _sampleIdx::Int
   """ legacy support for variable values old functor residual functions.
       TBD, this is still being used by DERelative factors. """
-  _legacyParams::X
+  _legacyParams::X #TODO rename to varValsHypo for consistent naming? and not not legacy any more
```

Comment:
One CalcFactor per sample?  History idea was CCW per solve and heap CalcFactor per sample.  Then we can do multithreading for each sample.  Also for more complicated factors that require own cache memory location.  Don't want to index into cf.somemem[sampleIdx]